utils/utils.ts

```diff
@@ -4,30 +4,81 @@
 
 interface IOptions {
   text: string;
-  options: Array<{text:string; correct: boolean}>;
+  options: Array<{ text: string; correct: boolean }>;
 }
 
-export function grade(optionsQuestion: Record<string, boolean>[], answers: IOptions[]) {
-  let notaAcumulada = 0;
-  let allQuestionsRight = true
+export function grade(
+  optionsQuestion: Record<string, boolean>[],
+  answers: IOptions[]
+) {
+  let finalGrade = 0;
+  let allQuestionsRight = true;
 
-  if (optionsQuestion.length !== answers.length) { return "error" }
+  console.log("user", optionsQuestion);
+  console.log("answers", answers);
+
+  if (optionsQuestion.length !== answers.length) {
+    return "error";
+  }
 
   for (let index = 0; index < optionsQuestion.length; index++) {
-    const answerUser = Object.values(optionsQuestion[index])
-    const answerServer = Object.values(answers[index].options)
+    const answerUser = Object.values(optionsQuestion[index]);
+    const answerServer = Object.values(answers[index].options);
 
     for (let indexObject = 0; indexObject < answerUser.length; indexObject++) {
       if (answerUser[indexObject] !== answerServer[indexObject].correct) {
-        allQuestionsRight = false
+        allQuestionsRight = false;
         break;
       }
-
     }
     if (allQuestionsRight) {
-      notaAcumulada++;
+      finalGrade++;
     }
   }
 
-  return `${notaAcumulada}/${answers.length}`
+  return `${finalGrade}/${answers.length}`;
+}
+
+/**
+ * Funcion para saber si la pregunta es multiple
+ */
+type Option = {
+  text: string;
+  correct: boolean;
+};
+
+export function isMultiple(options: Option[]) {
+  if (options === undefined) {
+    return true;
+  }
+
+  let count = 0;
+
+  for (let i = 0; i < options.length; i++) {
+    if (options[i].correct) count++;
+  }
+
+  return count > 1;
 }
+
+/**
+ * Preparar el estado inicial de las preguntas
+ */
+export function prepareInitialState(questionsServer: Option[], isMultiple: boolean) {
```

Comment:
here the second argument is redundant, you can use the questionsServer to check if they are multiple using the isMultiple function here

Reply:
mmmm maybe we can simplify this, with a more simple structure.